pkg/controllers/clusteroperator_controller.go

```diff
@@ -97,20 +97,8 @@ func (r *CloudOperatorReconciler) Reconcile(ctx context.Context, _ ctrl.Request)
 		return ctrl.Result{}, err
 	}
 
-	platform, err := config.GetProviderFromInfrastructure(infra)
-	if err != nil {
-		klog.Errorf("Unable to determine platform from infrastructure: %s", err)
-		// Ignoring error here as infrastructure resource needs to be reconciled externally
-		// to provide correct platform
-		if err := r.setStatusAvailable(ctx); err != nil {
-			klog.Errorf("Unable to sync cluster operator status: %s", err)
-			return ctrl.Result{}, err
-		}
-		return ctrl.Result{}, nil
-	}
-
 	// Verify FeatureGate ExternalCloudProvider is enabled for operator to work in TP phase
-	external, err := cloudprovider.IsCloudProviderExternal(platform, featureGate)
+	external, err := cloudprovider.IsCloudProviderExternal(infra.Status.PlatformStatus, featureGate)
```

Comment:
`infra.Status.PlatformStatus` Can this nil pointer deref?

Reply:
There is a check in library-go https://github.com/openshift/library-go/blob/master/pkg/cloudprovider/external.go#L21

Reply:
There is one above in compose config itself, and it does nil check.